browserdebuggertools/sockethandler.py

```diff
@@ -175,6 +181,18 @@ def execute(self, domain_name, method_name, params=None):
         })
         return self._wait_for_result()
 
+    def execute_async(self, domain_name, method_name, params=None):
```

Comment:
You mentioned already that execute_async needs more work. Fair enough.
So far (in case you haven't noticed), most of it can be commonised with execute.